### PR TITLE
Apptree 3819

### PR DESCRIPTION
**Pull request by SamOrozco**
Adding roles manager for adding, removing, and getting roles for appID.
Add User manager for querying a user based on given attributes.

https://apptree.myjetbrains.com/youtrack/issue/APPTREE-3819

### CODE REVIEW

File: app/sdk/roles/models/AppRole.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import sdk.annotations.PrimaryKey;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AppRole {
+    @PrimaryKey
+    @JsonProperty("roleName")
+    private String roleName;
+    @JsonProperty("roleID")
+    @Attribute(index = 0)
+    private int roleID;
```
Comment (karikittie) [inline app/sdk/roles/models/AppRole.java]: Is there any way for this to come back from the core null?

File: app/sdk/user/UserManager.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static sdk.utils.ValidationUtils.NullOrEmpty;
+
+public class UserManager {
+    private String programmaticUserToken;
+    private WSClient wsClient;
+    private ObjectMapper objectMapper;
+    private static final String queryUserRoute = "/public/1/users/query";
```
Comment (karikittie) [inline app/sdk/user/UserManager.java]: should this be in the constants class?

File: app/sdk/user/UserManager.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static sdk.utils.ValidationUtils.NullOrEmpty;
+
+public class UserManager {
+    private String programmaticUserToken;
+    private WSClient wsClient;
+    private ObjectMapper objectMapper;
```
Comment (karikittie) [inline app/sdk/user/UserManager.java]: should this be static?

File: app/sdk/roles/models/AppRole.java (hunk 1 above)
Comment (SamOrozco) [reply app/sdk/roles/models/AppRole.java]: @karikittie There shouldn't be.

File: app/sdk/user/UserManager.java (hunk 3 above)
Comment (SamOrozco) [reply app/sdk/user/UserManager.java]: No it's lazily loaded, so if role manager is never used it is never created.

File: app/sdk/user/UserManager.java (hunk 2 above)
Comment (SamOrozco) [reply app/sdk/user/UserManager.java]: @karikittie I wouldn't think so because this is the class that is managing it.

**Pull request closed without merging**